python-checks/src/main/java/org/sonar/python/checks/MandatoryFunctionReturnTypeHintCheck.java

```diff
@@ -55,6 +73,66 @@ private static void raiseIssueForConstructor(SubscriptionContext ctx, Name funct
     PythonQuickFix quickFix = PythonQuickFix.newQuickFix(CONSTRUCTOR_MESSAGE)
       .addTextEdit(TextEditUtils.insertAfter(functionDef.rightPar(), " -> None"))
       .build();
-    preciseIssue.addQuickFix(quickFix); 
+    preciseIssue.addQuickFix(quickFix);
+  }
+
+  private static void raiseIssueForReturnType(SubscriptionContext ctx, Name functionName, FunctionDef functionDef) {
+    PreciseIssue issue = ctx.addIssue(functionName, MESSAGE);
+    ReturnStatementVisitor returnStatementVisitor = new ReturnStatementVisitor();
+    functionDef.body().accept(returnStatementVisitor);
+    if (!returnStatementVisitor.returnStatements.isEmpty()) {
+      addQuickFixForReturnType(issue, functionDef, returnStatementVisitor.returnStatements);
+    } else {
+      if (returnStatementVisitor.yieldStatements.isEmpty()) {
+        addQuickFixForNoneType(issue, functionDef);
+      }
+    }
```

Comment:
_minor_ We can simplify this a bit:
```suggestion
    } else if (returnStatementVisitor.yieldStatements.isEmpty()) {
        addQuickFixForNoneType(issue, functionDef);
    }
```